io/las/LasReader.cpp

```diff
@@ -619,7 +619,17 @@ point_count_t LasReader::readFileBlock(std::vector<char>& buf,
     point_count_t blockpoints = buf.size() / ptLen;
 
     blockpoints = std::min(maxpoints, blockpoints);
+    if (m_istream->eof())
+        throw pdal::invalid_stream("stream is done");
```

Comment:
I threw that because the thing calling it was explicitly catching it and dropping it on the floor right away. 

This seemed the least disruptive, but if you have other ideas, please push them to this branch. 

> On May 8, 2015, at 11:54 AM, Andrew Bell <notifications@github.com> wrote:
> 
> In io/las/LasReader.cpp:
> 
> > @@ -619,7 +619,17 @@ point_count_t LasReader::readFileBlock(std::vector<char>& buf,
> >      point_count_t blockpoints = buf.size() / ptLen;
> >  
> >      blockpoints = std::min(maxpoints, blockpoints);
> > +    if (m_istream->eof())
> > +        throw pdal::invalid_stream("stream is done");
> Not sure I like throwing an exception here. Also, I've been trying to get rid of all the subclassed exceptions that we're not using anyway (of course I haven't TOLD anyone that I've been doing this.)
> 
> —
> Reply to this email directly or view it on GitHub.
>